addons/promote/common/src/main/java/org/commonjava/indy/promote/data/PathConflictManager.java

```diff
@@ -0,0 +1,86 @@
+package org.commonjava.indy.promote.data;
+
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.StoreKeyPaths;
+import org.commonjava.indy.promote.model.PathsPromoteResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import static java.util.Collections.disjoint;
+
+@ApplicationScoped
+public class PathConflictManager
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    private final Map<StoreKey, Set<StoreKeyPaths>> conflictsMap = new HashMap<>();
```

Comment:
I would oppose here that Indy should not be focused on PNC usage that much. We have to anticipate anyone can use it in a different way where it could cause issues. Also we can change the approach in PNC in the future and hit that too.

Reply:
Got it. I removed the empty set from map.